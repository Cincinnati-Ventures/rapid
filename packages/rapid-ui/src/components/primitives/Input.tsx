import * as React from "react"
import { createVariant } from "../../theme";
import { RapidStyles } from "../../utils";


const THEME_CLASSNAME = 'rapid-input';

export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
    styles?: string;
    variant?: string;
}

export const inputTheme = createVariant({
  baseStyle: 'rounded-lg transition-all ease-out duration-300 outline-none border border-lightGrey focus:outline-none focus:shadow-button-focus focus:border-main',
	variants: {
		default: 'px-3 py-1',
    filled: 'bg-lightGrey focus:bg-white',
    flushed: '',
    unstyled: '',
	},
	defaultProps: {
		variant: 'default',
	},
});

const getVariantClassName = (variant: string | undefined) => {
	if (variant) return `rapid-input-${variant}`;
	else return undefined;
}

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ styles, placeholder, variant, ...rest }, ref) => {
    return (
      <input
        ref={ref}
        {...rest}
        placeholder={placeholder || 'Rapid Input'}
        className={RapidStyles(styles, getVariantClassName(variant) || THEME_CLASSNAME)}
      />
    )
  }
)

Input.displayName = "Input"

export default Input;
