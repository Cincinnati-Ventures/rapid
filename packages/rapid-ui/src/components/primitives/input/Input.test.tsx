import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import Input from './Input';

describe('Input', () => {
	it('should render default correctly', () => {
		const { container } = render(<Input />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with variant', () => {
		const { container } = render(<Input variant='faded' />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with variant and custom styles', () => {
		const { container } = render(
			<Input styles='bg-green-200' variant='flushed' />,
		);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with variant, custom styles, and placeholder', () => {
		const { container } = render(
			<Input
				styles='bg-red-100'
				variant='invalid'
				placeholder='Placeholder text'
			/>,
		);
		expect(container).toMatchSnapshot();
	});
});
