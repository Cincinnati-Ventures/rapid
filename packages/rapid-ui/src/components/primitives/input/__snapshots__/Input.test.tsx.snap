// Vitest Snapshot v1

exports[`Input > should render correctly with variant 1`] = `
<div>
  <input
    class="rapid-input-faded"
    placeholder="Rapid Input"
  />
</div>
`;

exports[`Input > should render correctly with variant and custom styles 1`] = `
<div>
  <input
    class="rapid-input-flushed bg-green-200"
    placeholder="Rapid Input"
  />
</div>
`;

exports[`Input > should render correctly with variant, custom styles, and placeholder 1`] = `
<div>
  <input
    class="rapid-input-invalid bg-red-100"
    placeholder="Placeholder text"
  />
</div>
`;

exports[`Input > should render default correctly 1`] = `
<div>
  <input
    class="rapid-input"
    placeholder="Rapid Input"
  />
</div>
`;
